shardingsphere-infra/shardingsphere-infra-optimize/src/main/java/org/apache/shardingsphere/infra/optimize/context/CalciteContextFactory.java

```diff
@@ -54,68 +63,106 @@
                 .withIdentifierMaxLength(SqlParser.DEFAULT_IDENTIFIER_MAX_LENGTH)
                 .withConformance(connectionConfig.conformance())
                 .withParserFactory(SqlParserImpl.FACTORY);
         typeFactory = new JavaTypeFactoryImpl();
-        factory = new CalciteLogicSchemaFactory(metaDataMap);
         cluster = newCluster();
     }
-    
-    private Properties createProperties() {
-        // TODO Not only MySQL here.
-        Properties result = new Properties();
-        result.setProperty(CalciteConnectionProperty.LEX.camelName(), Lex.MYSQL.name());
-        result.setProperty(CalciteConnectionProperty.CONFORMANCE.camelName(), SqlConformanceEnum.MYSQL_5.name());
-        return result;
+
+    private void initProperties(final DatabaseType databaseType) {
+        // TODO Logic could be improved.
+        if (databaseType instanceof H2DatabaseType) {
+            // TODO No suitable type of Lex
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.MYSQL.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.DEFAULT.name());
+            return;
+        }
+        if (databaseType instanceof MariaDBDatabaseType) {
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.MYSQL.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.MYSQL_5.name());
+            return;
+        }
+        if (databaseType instanceof MySQLDatabaseType) {
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.MYSQL.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.MYSQL_5.name());
+            return;
+        }
+        if (databaseType instanceof OracleDatabaseType) {
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.ORACLE.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.ORACLE_12.name());
+            return;
+        }
+        if (databaseType instanceof PostgreSQLDatabaseType) {
+            // TODO No suitable type of Lex and conformance
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.MYSQL.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.DEFAULT.name());
+            return;
+        }
+        if (databaseType instanceof SQL92DatabaseType) {
+            // TODO No suitable type of Lex
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.MYSQL.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.STRICT_92.name());
+            return;
+        }
+        if (databaseType instanceof SQLServerDatabaseType) {
+            this.properties.setProperty(LEX_CAMEL_NAME, Lex.SQL_SERVER.name());
+            this.properties.setProperty(CONFORMANCE_CAMEL_NAME, SqlConformanceEnum.SQL_SERVER_2008.name());
+            return;
+        }
+        throw new ShardingSphereException("No matching DatabaseType found");
```

Comment:
I guess there is no any chance running here, right? Please consider removing it.

Reply:
In order to maintain the robustness of the code and avoid some unknown errors, please allow me to keep this part....